registry/test/apps.js

```diff
@@ -0,0 +1,388 @@
+const _ = require('lodash');
+const { expect } = require('chai');
+
+const example = {
+    url: '/api/v1/app/',
+    correct: Object.freeze({
+        name: '@portal/ncTestAppReactssr',
+        spaBundle: 'http://localhost:1234/ncTestAppReactssr.js',
+        cssBundle: 'http://127.0.0.1:1234/ncTestAppReactssr.css',
+        ssr: {
+            src: "http://127.0.0.1:1234/fragment",
+            timeout: 1000,
+            primary: true
+        },
+        assetsDiscoveryUrl: 'http://127.0.0.1:1234/_spa/dev/assets-discovery',
+        // dependencies: {},
+        // props: {},
+        // initProps: {},
+    }),
+    incorrect: Object.freeze({
+        name: 123,
+        spaBundle: 456,
+        cssBundle: 789,
+        ssr: 456,
+        assetsDiscoveryUrl: 789,
+        dependencies: 456,
+        props: 789,
+        initProps: 456,
+    }),
+    updated: Object.freeze({
+        name: '@portal/ncTestAppReactssr',
+        spaBundle: 'http://localhost:1234/ncTestAppReactssrUpdated.js',
+        cssBundle: 'http://127.0.0.1:1234/ncTestAppReactssrUpdated.css',
+        ssr: {
+            src: "http://127.0.0.1:1234/fragmentUpdated",
+            timeout: 2000,
+            primary: false,
+        },
+        assetsDiscoveryUrl: 'http://127.0.0.1:1234/_spa/dev/assets-discoveryUpdated',
+        dependencies: {
+            react: 'https://cdnjs.cloudflare.com/ajax/libs/react/16.8.6/umd/react.development.js',
+            'react-dom': 'https://cdnjs.cloudflare.com/ajax/libs/react-dom/16.8.6/umd/react-dom.development.js',
+        },
+        props: {
+            fragmentModuleName: 'reactssr-app-mainUpdated',
+            assetsPath: 'http://127.0.0.1:3001/uisamplereactUpdated',
+            locationStrategy: 'browserHistoryUpdated',
+        },
+        initProps: {
+            assetsPath: 'http://127.0.0.1:3001/uisamplereact',
+        },
+    }),
+
+    create: async () => {
+        await request.post(example.url).send(example.correct);
+    },
+    delete: async () => {
+        await request.delete(example.url + example.encodedName);
+    },
+};
+example.encodedName = encodeURIComponent(example.correct.name);
+
+describe(`Tests ${example.url}`, () => {
+    before('should work simple "create" and "delete"', async () => {
+        await request.post(example.url).send(example.correct).expect(200)
+        await request.get(example.url + example.encodedName).expect(200);
+        await request.delete(example.url + example.encodedName).expect(204);
+        await request.get(example.url + example.encodedName).expect(404);
+    });
+    afterEach(example.delete);
+    describe('Create', () => {
```

Comment:
In my opinion, this name of this test suite is not a good
The same is for all test suites and test cases
I guess it should describe it from a user perspective
So, for example, `when a user is creating a new app` so after a test case describes what should be in special case `should create something` or `should get a validation error` etc.